### PR TITLE
LOOP-1000 LOOP-990 LOOP-989 LOOP-987 Settings setup

### PR DESCRIPTION
**Pull request by jh-bate**
- flattened the structure
- addition of labels file so can more easily deal with other language in the future 
- expansion of risk tests 
- expansion of the smoke test
- common structure with add meal, bolus, exercise overrides and settings screens

### CODE REVIEW

File: e2e/smoke_test/smokeTest.spec.js [hunk 1]
```diff
@@ -76,75 +26,115 @@ describe('smoke test', () => {
...
-    });
-    describe('pump', () => {
-        it('can be added', async () => {
-            await pump.AddSimulator();
+        describe('cgm', () => {
+            it('can be added', async () => {
+                await settings.AddCGMSimulator();
+            });
+            it('can configure simulator model', async () => {
+                await settings.SetCGMModel(CGMModel.Constant, ['114']);
```
Comment (darinkrauss) [inline e2e/smoke_test/smokeTest.spec.js:53]: [nit] Consider handling simulator settings much like you do loop settings. After you add the simulator, then a command to go into the simulator settings, and a way to set each setting. Not needed for now, but if you find you are modifying the CGM/pump simulator settings frequently.

File: e2e/wip/settings.valid.spec.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        settings: SettingDefault,
+    };
+    it('we apply all settings', async () => {
+        await loopSettings.Configure(config);
+    });
+    it('should advance the scenario so we are looping', async () => {
+        await setup.AdvanceScenario(config.scenario, '1');
+        await waitFor(match.accessible.ButtonBarButton('Settings')).toExist().withTimeout(2000);
+    });
+    // it('should show no alert when tapping loop icon', async () => {
```
Comment (darinkrauss) [inline e2e/wip/settings.valid.spec.js:19]: [nit] Commented out?

File: src/bolus.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+     */
+    async SetAuthentication() {
+        await match.UITextField().clearText();
+        await match.UITextField().typeText(units);
+    }
+
+
+};
+
+module.exports = { Bolus };
```
Comment (darinkrauss) [inline src/bolus.js:45]: [nit] Don't forget the last newline on the file. Your editor might do this for you.

File: src/main.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        await match.UITextField().typeText(units);
+    }
+    /**
+     * @example await bolus.SetAuthentication();
+     */
+    async SetAuthentication() {
+        await match.UITextField().clearText();
+        await match.UITextField().typeText(units);
+    }
+
```
Comment (darinkrauss) [inline src/main.js:41]: [nit] Extra blank lines.

File: src/setup.js [hunk 5]
```diff
@@ -3,17 +3,15 @@ const match = require('./match');
 
 const setup = {
     /**
-     * @name LaunchLoop
      * @summary will launch the loop app with permissons for notifications and health enabled
      */
     async LaunchLoop() {
         await device.launchApp({
-            newInstance: true,
+            //newInstance: true,
```
Comment (darinkrauss) [inline src/setup.js:10]: [nit] Commented out?

File: e2e/regression_test/settings.no_basal_rates.spec.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+            scenario: 'flat_cgm_trace',
+            settings: FilterSettings(SettingDefault, [SettingType.BasalRates])
+        };
+        await loopSettings.Configure(config);
+    });
+    it('should not be in closed loop mode', async () => {
+        await expect(match.loop.Icon()).toHaveLabel('Waiting for first run');
+    });
+    it('should show error that indicates why not in closed loop mode', async () => {
+        await waitFor(match.loop.Icon()).toBeVisible().withTimeout(2000);
```
Comment (darinkrauss) [inline e2e/regression_test/settings.no_basal_rates.spec.js:18]: [nit] Per our discussion last week, for the future you may want to consider "hiding" this into a simple function to make it easier for others to code.

File: e2e/regression_test/settings.no_basal_rates.spec.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+describe('Closed loop is not allowed when settings', () => {
+    beforeAll(async () => {
+        await setup.LaunchLoop();
+    });
+    it('are not applied for basal rates', async () => {
+        let config = {
+            scenario: 'flat_cgm_trace',
+            settings: FilterSettings(SettingDefault, [SettingType.BasalRates])
+        };
+        await loopSettings.Configure(config);
```
Comment (darinkrauss) [inline e2e/regression_test/settings.no_basal_rates.spec.js:12]: [nit] Per our discussion last week, you can get rid of all of the extra `await`s.

**Review by darinkrauss — APPROVED** [review]
LGTM. A few nits for you to consider, but definitely nothing show stopping. 🚀

File: e2e/regression_test/settings.no_basal_rates.spec.js (hunk 6 above)
Comment (jh-bate) [reply e2e/regression_test/settings.no_basal_rates.spec.js:18]: agreed!

**Pull request merged**